content/shmem_team_split_2d.tex

```diff
@@ -0,0 +1,111 @@
+
+\apiargument{OUT}{yaxis\_team}{A new \ac{PE} team handle representing a \ac{PE}
+subset consisting of all the \acp{PE} that have the same coordinate along the Y-axis 
+as the calling \ac{PE}.}
+\end{apiarguments}
+
+\apidescription{
+The \FUNC{shmem\_team\_split\_2d} routine is a collective routine. It creates two new teams
+by splitting an existing parent team into up to two subsets based on a
+2D Cartesian space. The user provides the size of the X dimension, which is then used to
+derive the size of the Y dimension based on the size of the parent team. The size of
+the Y dimension will be equal to $ceiling ( N \div xrange )$, where \VAR{N} is the size
+of the parent team. In other words, $xrange \times yrange \geq N$, so that every \ac{PE}
+in the parent team has a unique \VAR{(x,y)} location the 2D Cartesian space.
+
+After the split operation, each of the new teams will contain all \acp{PE} that have the same
+coordinate along the X-axis and Y-axis, respectively, as the calling \ac{PE}. The
+\acp{PE} are numbered in the new teams based on the position of the \ac{PE} along
+the given axis.
+
+Any valid \openshmem team can be used as the parent team. This routine must be
+called by all \acp{PE} in the parent team. The value of \VAR{xrange} must be
+non-negative and all \acp{PE} in the parent team must pass the same value for
+\VAR{xrange}. None of the parameters need to reside in symmetric memory.
+
+Since the split may result in a 2D space with more points than there are members of
+the parent team, there may be a final, incomplete row of the 2D mapping of the parent
+team. This means that the resulting X-axis teams may vary in size by up to 1 \ac{PE},
+and that there may be one resulting Y-axis team of smaller size than all of the other
+Y-axis teams.
+
+The following grid shows the twelve teams that would result from splitting a parent team
+of size 10 with \VAR{xrange} of 3. The numbers in the grid cells are the \ac{PE} numbers
+in the parent team. The rows are the Y-axis teams. The columns are the X-axis teams.
+
+\begin{center}
+\begin{tabular}{|l|l|l|l|}
+ \hline
+      & x=0 & x=1 & x=2  \\ \hline
+ y=0  & 0   & 1   & 2  \\ \hline
+ y=1  & 3   & 4   & 5  \\ \hline
+ y=2  & 6   & 7   & 8  \\ \hline
+ y=3  & 9     \\
+ \cline{0-1} 
+\end{tabular}
+\end{center}
+
+Error checking will be done to detect if the value \LibConstRef{SHMEM\_TEAM\_NULL}
+is passed as the parent team handle. In that case, the program will abort with
+an informative error message. If an invalid team handle is passed to the
+routine, the behavior is undefined.
+
+The following options can be supplied during team split to restrict
+team functions and enable performance optimizations.  When using a given
+team, the application must comply with the requirements of all options
+set on that team; otherwise, the behavior is undefined.
+No options are enabled on \LibHandleRef{SHMEM\_TEAM\_WORLD} or on other
+predefined \openshmem teams.
```

Comment:
I think we need to say something to the effect of "all PEs in the resultant team must specify the same options". We have to be precise in the text here because, as noted earlier, the resultant X-axis and Y-axis teams don't need to have the same options at each other. It also seems like it should be valid that, for example, the Y-axis team where _y_ is zero might specify `NOCOLLECTIVE` while Y-axis teams where _y_ is not zero might not.

In other words, for all resultant teams from a call to `split_2d`, all PEs in the team must pass the same options for that team.

Reply:
Ah yes, that does make more sense. I will be more clear on this and maybe add a sentence to the example in the notes section.

Reply:
I am realizing this change needs to be made to the split_strided as well. I will put that in a different PR.